django/core/signing.py

```diff
@@ -74,7 +73,7 @@ def base64_hmac(salt, value, key):
 
 def get_cookie_signer(salt='django.core.signing.get_cookie_signer'):
     Signer = import_string(settings.SIGNING_BACKEND)
-    key = force_bytes(settings.SECRET_KEY)
+    key = settings.SECRET_KEY.encode()
```

Comment:
According to the [docs](https://docs.djangoproject.com/en/2.0/ref/settings/#std:setting-SECRET_KEY) and ticket https://code.djangoproject.com/ticket/24994 the type of SECRET_KEY can be either bytes or str.

> Uses of the key shouldn’t assume that it’s text or bytes. Every use should go through force_text() or force_bytes() to convert it to the desired type.

So a call to `.encode()` may not always work.

Reply:
so what is the suggested way?

Reply:
Unfortunately, I don't have a good answer for that.

I think now that Django is Python3 only, the project is in a better position to decide if SECRET_KEY is always either bytes or str. That would require some consensus from interested people, though.

As long as both types are allowed, some utility function is required. I guess at the moment that is force_bytes/force_text.